python/unityagents/environment.py

```diff
@@ -457,6 +457,11 @@ def _get_state(self, output: UnityRLOutput) -> (AllBrainInfo, bool):
             else:
                 [x.memories.extend([0] * (memory_size - len(x.memories))) for x in agent_info_list]
                 memory = np.array([x.memories for x in agent_info_list])
+            m_actions = np.ones((len(agent_info_list), self.brains[b].vector_action_space_size))
```

Comment:
Can you explain why you apply the mask here?

Reply:
This is where we create the mask. We initialize it with ones and then we add the zeros when appropriate. This is to avoid sending a bunch of ones when there is no mask.